src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -204,7 +204,8 @@ private static async Task<ImmutableArray<DiagnosticData>> GetProjectStateDiagnos
                 if (documentId != null)
                 {
                     // file doesn't exist in current solution
-                    var document = project.Solution.GetDocument(documentId);
+                    var document = project.Solution.GetDocument(documentId)
+                        ?? project.TryGetSourceGeneratedDocumentForAlreadyGeneratedId(documentId);
```

Comment:
Why not just call the async method?

Reply:
It may prevent a performance regression, but may introduce a functional regression if this is racing against another thread actually running generators, for example.

Stepping back for a second, what is this method doing, and how do we hit this in the regular document case? That comment on line 206 (not your doing, obviously) is 100% useless. :smile:

Reply:
⏱️ Will improve

Reply:
➡️ Now updated